



# Determining Success and Failure Factors for Startups
***

## Introduction

The purpose of this project is to find out the success and failure factors behind startup companies. To that end, I am going to investigate a dataset provided by [Metric.am](https://metric.am/), containing inforation about 472 startups and their status: *'Success'* or *'Failed'*. 

## Methodology

For the sake of this project, we will consider two classes of startups: *failed* and *successful*. As already mentioned above, we are interested in the factors behind the success or failure of the startup. If we tried to map the problem to the data, we would be interested in the importance of each feature or attribute while predicting the status of the company. Thus, it would be most meaningful and comfortable to solve the problem using a model which would provide us with this information. 

A model that seems suitable under the mentioned constraints is Logstic Regression. It is a classification algorithm, so it can trained to predict the status of the company. It also associates weights with idividual features, which we can use to determine the importance of the feature, as well as point out if its effect is positive or negative.

So this point onwards, we are going to pursue the following steps:
- [Retrieve the data](#Retrieving-data)
- [Explore the data](#Exploring-data)
- [Prepare the data]()
- [Build the classifier]()
    - [Split the data into train-test sets]()
    - [Build the model]()
    - [Train the model]()
- [Test the accuracy]()
- [Analyze the calculated weights]()
- [Report the conclusions](#Report)

***
## Retrieving data

First of all, let's download the data into a *.csv* file. I will use [wget](https://www.gnu.org/software/wget/) to download raw data from the [GitHub repository](https://github.com/Metricam/Internship_tasks/tree/master/Startup_Success).

In [11]:
!wget -O data.csv https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/data.csv

--2020-05-19 00:50:19--  https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/data.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.128.133, 151.101.192.133, 151.101.0.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.128.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 315563 (308K) [text/plain]
Saving to: ‘data.csv’

data.csv            100%[===================>] 308.17K  1.03MB/s    in 0.3s    

2020-05-19 00:50:20 (1.03 MB/s) - ‘data.csv’ saved [315563/315563]



We are also provided a *dictionary* dataset, which explains some of the features provided in the main dataset. Let's get that too.

In [12]:
!wget -O dictionary.csv https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/dictionary.csv

--2020-05-19 00:50:20--  https://raw.githubusercontent.com/Metricam/Internship_tasks/master/Startup_Success/dictionary.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.192.133, 151.101.0.133, 151.101.64.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.192.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5183 (5.1K) [text/plain]
Saving to: ‘dictionary.csv’

dictionary.csv      100%[===================>]   5.06K  --.-KB/s    in 0s      

2020-05-19 00:50:20 (12.7 MB/s) - ‘dictionary.csv’ saved [5183/5183]



***
## Exploring data

Now let's understand the data. For that we will need to import some libraries.

In [13]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

### About dataset

What are the attributes?

In [14]:
data_dict = pd.read_csv('dictionary.csv')
pd.options.display.max_rows = None
pd.options.display.max_columns = None
display(data_dict)

Variable  \
0                                         Company_Name   
1                             Dependent-Company Status   
2                                     year of founding   
3                              Age of company in years   
4                              Internet Activity Score   
5                 Short Description of company profile   
6                                  Industry of company   
7                           Focus functions of company   
8                                            Investors   
9                                       Employee Count   
10                          Employees count MoM change   
11                             Has the team size grown   
12                                  Est. Founding Date   
13                                   Last Funding Date   
14                                 Last Funding Amount   
15                                  Country of company   
16                                Continent of company   
17                         Number of Investors in Seed   
18              Number of Investors in Angel and or VC   
19                               Number of Co-founders   
20                               Number of of advisors   
21                         Team size Senior leadership   
22                             Team size all employees   
23   Presence of a top angel or venture fund in pre...   
24                       Number of of repeat investors   
25                   Number of  Sales Support material   
26                             Worked in top companies   
27    Average size of companies worked for in the past   
28             Have been part of startups in the past?   
29   Have been part of successful startups in the p...   
30          Was he or she partner in Big 5 consulting?   
31                              Consulting experience?   
32                         Product or service company?   
33        Catering to product/service across verticals   
34                    Focus on private or public data?   
35                             Focus on consumer data?   
36            Focus on structured or unstructured data   
37                         Subscription based business   
38             Cloud or platform based serive/product?   
39                              Local or global player   
40                 Linear or Non-linear business model   
41   Capital intensive business e.g. e-commerce, En...   
42                   Number of  of Partners of company   
43                        Crowdsourcing based business   
44                         Crowdfunding based business   
45                     Machine Learning based business   
46                       Predictive Analytics business   
47                           Speech analytics business   
48                     Prescriptive analytics business   
49                                   Big Data Business   
50         Cross-Channel Analytics/ marketing channels   
51   Owns data or not? (monetization of data) e.g. ...   
52   Is the company an aggregator/market place? e.g...   
53   Online or offline venture - physical location ...   
54                                 B2C or B2B venture?   
55   Top forums like 'Tech crunch' or 'Venture beat...   
56   Average Years of experience for founder and co...   
57                           Exposure across the globe   
58              Breadth of experience across verticals   
59                                   Highest education   
60                                  Years of education   
61                 Specialization of highest education   
62                   Relevance of education to venture   
63                  Relevance of experience to venture   
64          Degree from a Tier 1 or Tier 2 university?   
65                     Renowned in professional circle   
66         Experience in selling and building products   
67             Experience in Fortune 100 organizations   
68             Experience in Fortune 500 org

Let's take a look at our data. 

In [15]:
#df = pd.read_csv('data.csv')
#df.head()

The default encoder in read_csv() function is *UTF-8*, however, as we can see from the above error, the data in the *data.csv* file in not UTF-8 encoded. Let's try several other encodings to see if any of them succeeds to decode the data. Here are some: *latin1*, *iso-8859-1*, *cp1252*.

In [16]:
df = pd.read_csv('data.csv', encoding='latin1')
#df.head()
display(df)

Company_Name Dependent-Company Status year of founding  \
0       Company1                  Success          No Info   
1       Company2                  Success             2011   
2       Company3                  Success             2011   
3       Company4                  Success             2009   
4       Company5                  Success             2010   
5       Company6                  Success             2010   
6       Company7                  Success             2011   
7       Company8                  Success             2010   
8       Company9                  Success             2011   
9      Company10                  Success             2013   
10     Company11                  Success             2011   
11     Company12                  Success             2010   
12     Company13                  Success             2010   
13     Company14                  Success             2008   
14     Company15                  Success             2010   
15     Company16                  Success             2008   
16     Company17                  Success             2010   
17     Company18                  Success             2009   
18     Company19                  Success             2002   
19     Company20                  Success             2010   
20     Company21                  Success             2010   
21     Company22                  Success             2005   
22     Company23                  Success             2012   
23     Company24                  Success             2009   
24     Company25                  Success             2012   
25     Company26                  Success             2011   
26     Company27                  Success             2009   
27     Company28                  Success             2013   
28     Company29                  Success             2007   
29     Company30                  Success             2012   
30     Company31                  Success             2012   
31     Company32                  Success             2008   
32     Company33                  Success             2010   
33     Company34                  Success             2011   
34     Company35                  Success             2010   
35     Company36                  Success             2012   
36     Company37                  Success             2012   
37     Company38                  Success             2012   
38     Company39                  Success             2010   
39     Company40                  Success             2010   
40     Company41                  Success             2012   
41     Company42                  Success             2009   
42     Company43                  Success             2011   
43     Company44                  Success             2008   
44     Company45                  Success             2012   
45     Company46                  Success             2010   
46     Company47                  Success             2011   
47     Company48                  Success             2003   
48     Company49                  Success             2000   
49     Company50                   Failed          No Info   
50     Company51                  Success             2011   
51     Company52                  Success             2009   
52     Company53                  Success             2009   
53     Company54                  Success             2007   
54     Company55                  Success             2012   
55     Company56                  Success             2011   
56     Company57                  Success             2008   
57     Company58                  Success             2012   
58     Company59                  Success             2012   
59     Company60                  Success             2012   
60     Company61                  Success             2009   
61     Company62                  Success             2012   
62     Company63                  Success             2012   
63     Company64                  Su

Lucky for us, *latin1* was the real encoding!

***
## Preparing data

Now let's prepare our data for classification and separate the features from the labels. 

In [17]:
print("Shape before cleaning: ", df.shape)

Shape before cleaning:  (472, 116)


### Choosing the feature set...

First of all, it seems like following columns don't give too much value for our purpose: *'Company_Name', 'Short Description of company profile', 'Last Funding Date', 'Est. Founding Date', 'Age of company in years', 'Gartner hype cycle stage', 'Time to maturity of technology (in years)'*, so let's drop them!

Also, the status column should not be in the feature set.

In [18]:
features = df.drop(['Dependent-Company Status', 'Company_Name', 'Short Description of company profile', 
                    'Last Funding Date', 'Est. Founding Date', 'Age of company in years', 
                    'Gartner hype cycle stage', 'Time to maturity of technology (in years)'], axis=1)
features.head()

year of founding  Internet Activity Score  \
0          No Info                     -1.0   
1             2011                    125.0   
2             2011                    455.0   
3             2009                    -99.0   
4             2010                    496.0   

                              Industry of company Focus functions of company  \
0                                             NaN                  operation   
1          Market Research|Marketing|Crowdfunding           Marketing, sales   
2  Analytics|Cloud Computing|Software Development                 operations   
3                                Mobile|Analytics          Marketing & Sales   
4         Analytics|Marketing|Enterprise Software          Marketing & Sales   

                                           Investors  Employee Count  \
0  KPCB Holdings|Draper Fisher Jurvetson (DFJ)|Kl...             3.0   
1                                                NaN             NaN   
2  TechStars|Streamlined Ventures|Amplify Partner...            14.0   
3  Michael Birch|Max Levchin|Sequoia Capital|Keit...            45.0   
4  DFJ Frontier|Draper Nexus Ventures|Gil Elbaz|A...            39.0   

   Employees count MoM change Has the team size grown  Last Funding Amount  \
0                         0.0                      No             450000.0   
1                         NaN                      No                  NaN   
2                         0.0                      No            2350000.0   
3                        10.0                      No           10250000.0   
4                         3.0                      No            5500000.0   

  Country of company Continent of company Number of Investors in Seed  \
0      United States        North America                           2   
1      United States        North America                           5   
2      United States        North America                          15   
3      United States        North America                           6   
4      United States        North America                           7   

  Number of Investors in Angel and or VC  Number of Co-founders  \
0                                      0                      1   
1                                      0                      2   
2                                      0                      3   
3                                      0                      2   
4                                      0                      1   

   Number of of advisors  Team size Senior leadership Team size all employees  \
0                      2                            2                      15   
1                      0                            4                      20   
2                      0                            7                      10   
3                      0                            4                      50   
4                      1                            8                      40   

  Presence of a top angel or venture fund in previous round of investment  \
0                                                Yes                        
1                                                 No                        
2                                                 No                        
3                                                Yes                        
4                                                 No                        

  Number of of repeat investors Number of  Sales Support material  \
0                             4                           Nothing   
1                             0                            medium   
2                             0                               low   
3                             0                               low   
4                             0                              high   

  Worked in top companies Average size of companies worked for in the past  \
0                      No                          

### Handling NaN values...

Before we rush to drop the rows that have at least one NaN value, let's see which features have the most NaN values. Maybe the presence of those features is less important compared to the amount of data we'll lose...

In [19]:
nan_df = pd.DataFrame(features.isna().sum().sort_values(ascending=False))
print(nan_df[nan_df[0] != 0].count())
nan_df[nan_df[0] != 0]

0    12
dtype: int64


0
Employees count MoM change           205
Employee Count                       166
Last Funding Amount                  160
Investors                            140
Industry of company                  124
Specialization of highest education   97
Industry trend in investing           82
Continent of company                  71
Country of company                    71
Internet Activity Score               65
Has the team size grown               50
Focus functions of company            30

The overall picture is not that bad, from 116 features, only 12 have NaN values, the rest are either fully filled in or the NaN values are expressed verbally, like 'no info'. Let's now quickly go over the list and see if it makes sense to keep these features.

- **'Employees count MoM change'** makes nearly the half of the dataset drop, so let's just get rid of that feature, as I don't even know what MoM is (Ministry of Manpower?).
- **'Employee Count'** who doesn't have this record? It seems like such a  primitive question! Anyways, this is an important data, and, sadly, we have to drop 166 lines!
- **'Last Funding Amount'** let's just drop this column.
- **'Investors'** this one is different, my intuition is that the rows that don't have investors have NaN values, so instead of dropping them, I'm going to substitute those with empty strings. Later, when I will make this variable binary, those rows will just have 0s for all the Investors. 
- **'Industry of company'** is pretty important in my opinion, so I'm going to drop the rows that have NaN.
- **'Specialization of highest education'** is a crazy field: some have meaningful answers and some have things like _"PhD"_ or _"INDUSTRI"_ or _"computers"_. Plus, this field is not directly connected to the success, in my opinion, the connection between the field of the startup and the specialization is what matters. We could filter all the fields out from specialization in complex ways and then compare with the fields of the startup, but if we take a quick look at the dataset, we'll see that the answers vary a lot and maybe it's best in this case to just discard this column and not over-engineer.
- The rest are fairly important and don't have drastic results, so we'll just drom the NaN rows.

In [20]:
# Drop the discussed columns.
features = features.drop(['Employees count MoM change', 'Last Funding Amount', 
                          'Specialization of highest education'], axis=1)

# Sustitute NaNs with empty strings.
features['Investors'] = features['Investors'].apply(lambda x: x if isinstance(x, str) else '')

# Drop the rest of the rows.
features = features.dropna()

features.head()

year of founding  Internet Activity Score  \
2             2011                    455.0   
3             2009                    -99.0   
4             2010                    496.0   
5             2010                    106.0   
6             2011                     39.0   

                              Industry of company Focus functions of company  \
2  Analytics|Cloud Computing|Software Development                 operations   
3                                Mobile|Analytics          Marketing & Sales   
4         Analytics|Marketing|Enterprise Software          Marketing & Sales   
5                    Food & Beverages|Hospitality                  analytics   
6                                       Analytics                   Research   

                                           Investors  Employee Count  \
2  TechStars|Streamlined Ventures|Amplify Partner...            14.0   
3  Michael Birch|Max Levchin|Sequoia Capital|Keit...            45.0   
4  DFJ Frontier|Draper Nexus Ventures|Gil Elbaz|A...            39.0   
5  Pritzker Group Venture Capital|Excelerate Labs...            14.0   
6  Plug & Play Ventures|Correlation Ventures|Cros...             7.0   

  Has the team size grown Country of company Continent of company  \
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   
5                      No      United States        North America   
6                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
2                          15                                      0   
3                           6                                      0   
4                           7                                      0   
5                           2                                      0   
6                           7                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
2                      3                      0                            7   
3                      2                      0                            4   
4                      1                      1                            8   
5                      4                      0                            4   
6                      2                      9                            2   

  Team size all employees  \
2                      10   
3                      50   
4                      40   
5                      14   
6                      15   

  Presence of a top angel or venture fund in previous round of investment  \
2                                                 No                        
3                                                Yes                        
4                                                 No                        
5                                                 No                        
6                                                 No                        

  Number of of repeat investors Number of  Sales Support material  \
2                             0                               low   
3                             0                               low   
4                             0                              high   
5                             2                            medium   
6                             4                            medium   

  Worked in top companies Average size of companies worked for in the past  \
2                     Yes                                           Medium   
3                      No                                            Large   
4                      No                                            Small   
5                      No                                           Medium   
6                      No        

In [21]:
features.shape

(239, 105)

### Converting list values into binary variables...

Let's now convert the following values into lists, so that we can then make them binary variables.

 - *'Industry of company'*
 - *'Investors'*
 - *'Focus functions of company'*

In [22]:
import re

features['Industry of company'] = features['Industry of company'].apply(lambda x: x.lower().strip().split('|'))
features['Investors'] = features['Investors'].apply(lambda x: x.lower().strip().split('|'))

def process_str(x) :
    str_list = re.split(',|&|\n|\|', x.lower())
    str_list = list(map(str.strip, str_list))
    return str_list

features['Focus functions of company'] = features['Focus functions of company'].apply(lambda x: process_str(x))

features.head()

year of founding  Internet Activity Score  \
2             2011                    455.0   
3             2009                    -99.0   
4             2010                    496.0   
5             2010                    106.0   
6             2011                     39.0   

                                 Industry of company  \
2  [analytics, cloud computing, software developm...   
3                                [mobile, analytics]   
4        [analytics, marketing, enterprise software]   
5                    [food & beverages, hospitality]   
6                                        [analytics]   

  Focus functions of company  \
2               [operations]   
3         [marketing, sales]   
4         [marketing, sales]   
5                [analytics]   
6                 [research]   

                                           Investors  Employee Count  \
2  [techstars, streamlined ventures, amplify part...            14.0   
3  [michael birch, max levchin, sequoia capital, ...            45.0   
4  [dfj frontier, draper nexus ventures, gil elba...            39.0   
5  [pritzker group venture capital, excelerate la...            14.0   
6  [plug & play ventures, correlation ventures, c...             7.0   

  Has the team size grown Country of company Continent of company  \
2                      No      United States        North America   
3                      No      United States        North America   
4                      No      United States        North America   
5                      No      United States        North America   
6                      No      United States        North America   

  Number of Investors in Seed Number of Investors in Angel and or VC  \
2                          15                                      0   
3                           6                                      0   
4                           7                                      0   
5                           2                                      0   
6                           7                                      0   

   Number of Co-founders  Number of of advisors  Team size Senior leadership  \
2                      3                      0                            7   
3                      2                      0                            4   
4                      1                      1                            8   
5                      4                      0                            4   
6                      2                      9                            2   

  Team size all employees  \
2                      10   
3                      50   
4                      40   
5                      14   
6                      15   

  Presence of a top angel or venture fund in previous round of investment  \
2                                                 No                        
3                                                Yes                        
4                                                 No                        
5                                                 No                        
6                                                 No                        

  Number of of repeat investors Number of  Sales Support material  \
2                             0                               low   
3                             0                               low   
4                             0                              high   
5                             2                            medium   
6                             4                            medium   

  Worked in top companies Average size of companies worked for in the past  \
2                     Yes                                           Medium   
3                      No                                            Large   
4                      No                                            Small   
5                      No                                     

### Encoding categorical variables as binary variables...

### Converting dates into *date time object*...

### Normalizing the data...

***
## Classification

### Train-test split

### Building the model

### Training

***
## Testing accuracy

***
## Analyzing results

***
## Report

***
## Thank you!

**Author:** [Aneta Baloyan](https://www.linkedin.com/in/aneta-baloyan/)

Email: *aneta.baloyan@gmail.com*

***

May 2020

